pypeit/core/gui/edge_inspector.py

```diff
@@ -0,0 +1,321 @@
+                                                      self.image_slider)
+        # And the button that will toggle the images
+        self.image_change_button = widgets.Button(image_change_button_ax, 'Image', color='0.7',
+                                                  hovercolor='0.9')
+        self.image_change_button.on_clicked(self.image_updater.next_image)
+        # And the button that will update the Edges object (rm, add, sync
+        # traces) and update the line plot
+        self.sync_button = widgets.Button(sync_button_ax, 'Sync', color='0.7', hovercolor='0.9')
+        self.sync_button.on_clicked(self.update_traces)
+
+        # Initialize and plot the traces
+        self._reset_traces()
+        self.plot_traces()
+
+        # Create the pointer
+        self.img_pointer = gui_util.Pointer(self.image_plot.axes, name='image', horizOn=False,
+                                            color='C1', lw=0.5)
+        # Add its functions
+        self.img_pointer.register('m', self.move, descr='Move the nearest trace to the cursor')
+        self.img_pointer.register('d', self.delete, descr='Delete the nearest trace')
+        self.img_pointer.register('l', self.add_left, descr='Add a left trace')
+        self.img_pointer.register('r', self.add_right, descr='Add a right trace')
+        self.img_pointer.register('U', self.undo, descr='Undo all changes since last sync')
+        # Add the help dialog
+        self.img_pointer.build_help()
+
+    def close(self):
+        """
+        Close the GUI.  Restores matplotlib RC defaults and closes the figure.
+        """
+        pyplot.rcdefaults()
+        self.fig.clear()
+        pyplot.close(self.fig)
+
+    def _trace_color(self, side):
+        """
+        Return the color to use for each edge sided.
+        
+        Args:
+            side (:obj:`int`):
+                Edge side.  Either -1 for left or 1 for right.
+
+        Returns:
+            :obj:`str`: The color to use for the trace.
+        """
+        return 'C2' if side < 0 else 'C4'
+
+    def plot_traces(self):
+        """
+        Plot the traces from scratch.
+        """
+        self.trace_plot = []
+        for i in range(self.ntrace):
+            if not self.good[i]:
+                self.trace_plot += [None]
+                continue
+            color = self._trace_color(self.side[i])
+            # Trace data should *not* be masked so that calling plot only
+            # produces a single Line2D object
+            self.trace_plot += [self.image_plot.axes.plot(self.trace_cen[:,i], self.spec_pix,
+                                                          color=color, lw=2)[0]]
+
+    def update_traces(self, *args, **kwargs):
+        """
+        Update the underlying trace data and edges object.
+
+        Changes to the traces are kept by the internals until we're ready to
+        "update" the edges object.  This performs the update and replots the
+        trace data.
```

Comment:
I added some explanation here for why this function has *args and **kwargs.  I originally only used `self` in the arguments, but the code barfed because the Button was passing something to the function.  Adding the placeholders that are always ignored is really just a lazy way out.  I removed **kwargs because it wasn't necessary.